url9d1719148e_mrqsdavi_pegmachine_tgz-pJ8-peg_RegexJ8/src/peg/Regex.java

```diff
@@ -0,0 +1,616 @@
+            iBackCommit.setInstrucaoDesvio(next);
+            
+            next = iBackCommit;
+
+            ArrayList<Instrucao> instrucoesRepeticao = instrucoesDoPadrao(padrao.e().getPadrao());
+
+            retorno.add(iChoice);
+            retorno.addAll(instrucoesRepeticao);
+            retorno.add(iBackCommit);
+            retorno.add(iFail);
+        }
+			break;
+		
+		case NAO:{
+
+            IChoice iChoice = new IChoice("");
+            ICommit iCommit = new ICommit("");
+            IFail iFail = new IFail();
+            IFailTwice iFailTwice = new IFailTwice();
+
+            iChoice.setInstrucaoDesvio(next);
+            iCommit.setInstrucaoDesvio(iFail);
+            
+            if(useFailTwiceOptimization){
+            	
+            	next = iFailTwice;
+
+                ArrayList<Instrucao> instrucoesRepeticao = instrucoesDoPadrao(padrao.nao().getPadrao());
+
+                retorno.add(iChoice);
+                retorno.addAll(instrucoesRepeticao);
+                retorno.add(iFailTwice);
+            	
+            }else{
+            	
+            	next = iFail;
+
+                ArrayList<Instrucao> instrucoesRepeticao = instrucoesDoPadrao(padrao.nao().getPadrao());
+
+                retorno.add(iChoice);
+                retorno.addAll(instrucoesRepeticao);
+                retorno.add(iCommit);
+                retorno.add(iFail);
+            	
+            }
+
+        }
+			break;
+			
+		case OPCIONAL:{		
+			
+			IChoice iChoice = new IChoice("");
+			ICommit iCommit = new ICommit("");
+			
+			iChoice.setInstrucaoDesvio(next);
+			iCommit.setInstrucaoDesvio(next);
+			
+			next = iCommit;
+			
+			ArrayList<Instrucao> instrucoesRepeticao = instrucoesDoPadrao(padrao.opcional().getPadrao());
+			
+			retorno.add(iChoice);
+			retorno.addAll(instrucoesRepeticao);
+			retorno.add(iCommit);
+			
+        }
+			break;
+			
+		case ATE:
+			retorno.add(new IFind(padrao.ate().getPadrao().conjunto().getConjuntoCaracteres()));
+			break;
+		
+		case CADEIA_VAZIA:
+			retorno.add(new IChar(true));
+			break;
+			
+		case SELF:
+			retorno.add(new ICall(labelUltimaGramatica));
+			break;
+		
+		case CHAMADA:{
+			ICall call = new ICall("");
+			call.setInstrucaoDesvio(instrucaoGramaticaTemporaria.get(padrao.chamada().getLabel()));
+			retorno.add(call);
+		}
+			break;
+			
+		case SEQUENCIA:
+			String texto = padrao.sequencia().getTexto();
+			for(int i = 0; i < texto.length(); i++){					
+				retorno.add(new IChar(texto.charAt(i)));
+			}
+			break;
+			
+		case PONTO:
+			retorno.add(new IAny(padrao.ponto().getNumero()));
+			break;
+			
+		case CONJUNTO:
+			retorno.add(new ICharset(padrao.conjunto().getTexto(), padrao.conjunto().getConjuntoCaracteres()));
+			break;
+		
+		default:
+			break;
+		}
+		
+		next = backupNext;
+		previous = backupPrevious;
+		
+		return retorno;		
+	}
+	
+	public void imprimirInstrucoes(ArrayList<Instrucao> instrucoes, HashMap<Instrucao, String> labelsIntrucao){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 59 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.